warehouse/query-core/src/main/java/datawave/query/predicate/EventDataQueryFieldFilter.java

```diff
@@ -93,18 +95,39 @@ public boolean keep(Key k) {
         return true;
     }
 
-    public KeyProjection getProjection() {
-        return keyProjection;
+    @Override
+    public boolean apply(@Nullable Map.Entry<Key,String> entry) {
+        if (entry == null) {
+            return false;
+        }
+        return apply(entry.getKey());
     }
 
     @Override
-    public boolean apply(@Nullable Map.Entry<Key,String> input) {
-        return keyProjection.apply(input);
+    public boolean peek(@Nullable Map.Entry<Key,String> entry) {
+        // equivalent to apply in the event column case, simple redirect
+        return apply(entry);
```

Comment:
peeking the same field repeatedly max falsely drive up the nextCount

Reply:
oooooh good point. I'll rework this.